Add replacement examples to getting_started.md (#2108)
JuliaData/DataFrames.jl: docs/src/man/getting_started.md

The exact rules of handling columns of a `DataFrame` are explained in
[The design of handling of columns of a `DataFrame`](@ref man-columnhandling) section of the manual.

## Replacing Data

> Use the broadcasting syntax to apply replacement operations to data frames.

Reviewer: It would be useful to specify that this is to replace values in all columns of a data frame. While you're at it, can you show an example replacing values in a single column? Two solutions are possible: using `ifelse` or using `replace!` (which is a bit shorter).

> Use the broadcasting syntax to apply replacement operations to `DataFrames`.

Reviewer 2: ```suggestion
Use the broadcasting syntax to apply replacement operations to data frames.
```

> Note: in-place replacement in a data frame requires the type of the replacement value match the respective column data type.

Reviewer: I'd first show a simple case which doesn't require `allowmissing!`, with an example code block, and then the more complex case.

> Note: in-place replacement in a `DataFrame` requires the type of the replacement value match the respective column data type.

Reviewer 2: ```suggestion
Note: in-place replacement in a data frame requires the type of the replacement value match the respective column data type.
```

> It follows that replacing `"None"` with `missing` requires a call to `allowmissing!(df)`.

Reviewer 2: ```suggestion
It follows that replacing `"None"` with `missing` in-place requires a call to `allowmissing!(df)` in the example below.
```

```jldoctest
julia> df = DataFrame(a = ["a", "None", "b", "None"], b = [1,2,3,4], c = ["None", "x", "y", "z"])
4×3 DataFrame
│ Row │ a       │ b      │ c       │

> │     │ String⍰ │ Int64⍰ │ String⍰ │

Reviewer 2: this is incorrect. at this stage `df` does not allow missing values

├─────┼────────┼───────┼────────┤
│ 1   │ a      │ 1     │ None   │
│ 2   │ None   │ 2     │ x      │
│ 3   │ b      │ 3     │ y      │
│ 4   │ None   │ 4     │ z      │

> julia> @. df = ifelse(df == "None", "c", df)

Reviewer: Using `@.` could confuse newcomers, better show the simpler syntax:
```suggestion
julia> df = ifelse.(df .== "None", "c", df)
```
Same below.

* In-place replacement in a data frame requires the type of the replacement value to match the respective column data type.
* It follows that replacing `"None"` with `missing` in-place requires a call to `allowmissing!(df)` in the final example below.

```jldoctest
julia> df = DataFrame(a = ["a", "None", "b", "None"], b = [1,2,3,4], c = ["None", "j", "k", "h"], d = ["x", "y", "None", "z"])
4×4 DataFrame
│ Row │ a      │ b     │ c      │ d      │
│     │ String │ Int64 │ String │ String │
├─────┼────────┼───────┼────────┼────────┤
│ 1   │ a      │ 1     │ None   │ x      │
│ 2   │ None   │ 2     │ j      │ y      │
│ 3   │ b      │ 3     │ k      │ None   │
│ 4   │ None   │ 4     │ h      │ z      │

# individual column
julia> replace!(df.a, "None"=> "c")
4-element Array{String,1}:
 "a"
 "c"
 "b"
 "c"

julia> df
4×4 DataFrame
│ Row │ a      │ b     │ c      │ d      │
│     │ String │ Int64 │ String │ String │
├─────┼────────┼───────┼────────┼────────┤
│ 1   │ a      │ 1     │ None   │ x      │
│ 2   │ c      │ 2     │ j      │ y      │
│ 3   │ b      │ 3     │ k      │ None   │
│ 4   │ c      │ 4     │ h      │ z      │

# replacement on a subset of columns [:c, :d]

> julia> df[:, [:c, :d]] = ifelse.(df[:, [:c, :d]] .== "None", "c", df[:, [:c, :d]])

Reviewer 2: no need to copy the data on RHS
```suggestion
julia> df[:, [:c, :d]] = ifelse.(df[:, [!c, :d]] .== "None", "c", df[!, [:c, :d]])
```
